### PR TITLE
Add C API bindings

### PR DESCRIPTION
**Pull request by jonlamb-gh**

### CODE REVIEW

File: client-libraries/c/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[package]
+name = "modality-capi"
+version = "0.1.0"
+edition = "2021"
+authors = ["Jon Lamb <jon@auxon.io>"]
+license = "Apache-2.0"
+repository = "https://github.com/auxoncorp/modality-sdk"
+description = "Part of Auxon's Modality SDK"
```
Comment (azdle) [inline client-libraries/c/Cargo.toml:8]: Heh, I just removed this description from all the rust libs. Can we:

```suggestion
description = "The C & C++ FFI for Auxon's Modality SDK"
```

File: client-libraries/c/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# libmodality_client C API
```
Comment (azdle) [inline client-libraries/c/README.md:1]: Would be nice to get a little bit of description in here. Maybe point to the `ctest` dir to see example use.

**Review by azdle — APPROVED** [review]
LGTM

Two nits inline below.

Also should add this to the list of implementations in readme in the repo root.

File: client-libraries/c/Cargo.toml (hunk 1 above)
Comment (jonlamb-gh) [reply client-libraries/c/Cargo.toml:8]: Sounds good, will do

File: client-libraries/c/README.md (hunk 2 above)
Comment (jonlamb-gh) [reply client-libraries/c/README.md:1]: Good call, will do

Comment (jonlamb-gh) [reply client-libraries/c/README.md]: Ended up deleting the README for now, will come back with a docs pass if needed

**Pull request merged**